openverse_catalog/dags/providers/provider_api_scripts/brooklyn_museum.py

```diff
@@ -2,186 +2,162 @@
+            return response_json.get("data")
+        return None
+
+    def get_should_continue(self, response_json):
+        # We can assume that any truthy response from the get data function would
+        # indicate we can continue
+        return bool(self._get_data_from_response(response_json))
+
+    def get_batch_data(self, response_json) -> list | None:
+        return self._get_data_from_response(response_json)
+
+    @staticmethod
+    def _get_license_url(rights_info):
+        elements = html.fromstring(rights_info.get("description", ""))
+        cc_links = [
+            link
+            for _, _, link, _ in elements.iterlinks()
+            if "https://creativecommons.org/" in link
+        ]
+        license_url = cc_links[0] if len(cc_links) == 1 else None
+        return license_url
+
+    @staticmethod
+    def _get_image_sizes(image):
+        height, width = None, None
+        size_list = image.get("derivatives", "")
+        if isinstance(size_list, list):
+            size_type = image.get("largest_derivative", "")
+            for size in size_list:
+                if size.get("size", "") == size_type:
+                    height = size.get("height")
+                    width = size.get("width")
+        return height, width
+
+    @staticmethod
+    def _get_metadata(data):
+        return {
+            "accession_number": data.get("accession_number"),
+            "date": data.get("object_date"),
+            "description": data.get("description"),
+            "medium": data.get("medium"),
+            "credit_line": data.get("credit_line"),
+            "classification": data.get("classification"),
+        }
+
+    @staticmethod
+    def _get_creators(data):
+        artists_info = data.get("artists")
+        if isinstance(artists_info, list):
+            creators_list = (
+                artists.get("name")
+                for artists in artists_info
+                if artists.get("rank") == 1
+            )
+            creator = next(creators_list, None)
+        else:
+            creator = None
+        return creator
+
+    @staticmethod
+    def _get_image_url(image):
+        image_url = image.get("largest_derivative_url")
+        if image_url and not image_url.startswith("http"):
+            image_url = "https://" + image_url
+        return image_url
+
+    @staticmethod
+    def _handle_object_data(data, license_url) -> list[dict]:
+        images = []
+        image_info = data.get("images")
+        if image_info is not None:
+            id_ = data.get("id", "")
+            title = data.get("title", "")
+            foreign_url = f"https://www.brooklynmuseum.org/opencollection/objects/{id_}"
+            metadata = BrooklynMuseumDataIngester._get_metadata(data)
+            creators = BrooklynMuseumDataIngester._get_creators(data)
+
+            for image in image_info:
+                foreign_id = image.get("id", "")
```

Comment:
I totally thought that `foreign_id` was required. Does this mean it's not?

Reply:
That's a very good point... 😱 I'll add some extra conditions here, we should just `continue` if `foreign_id` isn't found.